### DISCUSSION TITLE
Accessing to computer files from mod

### DISCUSSION THREAD

**Discussion by SirEdvin** [cc-tweaked/CC-Tweaked#1933] [Q&A] [answered]
Is there is a way now to access computer files from java code?
I used `serverComputer.computer.environment.fileSystem`, but now `computer` field is private ...

I basically need just read all files from one directory

**Comment by SquidDev** [comment]
> I basically need just read all files from one directory

What are you actually trying to do?

**Reply by SquidDev** [reply]
Ahh, have found [the file in question](https://github.com/SirEdvin/UnlimitedPeripheralWorks/blob/6efb33b67104c4c38edb189e397aa15948ba883b/projects/forge/src/main/kotlin/site/siredvin/peripheralworks/integrations/integrateddynamics/Integration.kt). No, there is no way to do that, and I'm not sure it'd be desirable? This feels like it should be a peripheral, or at least an API or something. Not configured through the file system!

**Reply by SirEdvin** [reply]
Probably, but to make it work correctly I need someway to attach meta information to computer and I come up with filesystem hook. Theoretically, I probably, should just have another SavedData instance, that would take care of this logic, rather then using filesystem and then just provide regular peripheral

**Reply by SquidDev** [reply, marked as answer]
Does it actually need to be persisted? My gut feeling is that there's no harm if this state is reset on chunk (un)load, but definitely don't have all the context here.

To be clear, I'm not going to stop you keeping the current behaviour (you can just use reflection to get the field). Just ... well, you know my opinion on using CC's internal classes :p.

**Reply by SirEdvin** [reply]
Probably you are right, it would be simpler for me to make it reset on server restarts rather than making it persistent and there is no harm done doing so.
